### PR TITLE
Add pre-commit that strips out cells

### PR DESCRIPTION
**Pull request by jsignell**
This just adds the pre-commit as suggested by #220 - it would be good to also have a github action that enforces the pre-commit.

### CODE REVIEW

**Review by jrbourbeau — COMMENTED** [review]
+1 thanks @jsignell

> it would be good to also have a github action that enforces the pre-commit.

Agreed. Would you mind adding that workflow here? I imagine we can just reuse what's already over in `dask/dask`

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+repos:
+  - repo: https://github.com/asottile/pyupgrade
+    rev: v2.31.0
+    hooks:
+      - id: pyupgrade
+        args:
+          - --py38-plus
```
Comment (jsignell) [inline .pre-commit-config.yaml:7]: I'm kind of of the opinion the more the merrier with pre-commit hooks and this one was easy to add.

Comment (jrbourbeau) [reply .pre-commit-config.yaml:7]: +1. We're also already using this pre-commit in `dask/dask` and `dask/distributed` -- so I like the consistent code styling

Comment (jsignell) [reply .pre-commit-config.yaml:7]: right. We can add black and things when we are feeling more ambitious :)

**Review by jrbourbeau — APPROVED** [review]

**Pull request merged**